### ISSUE TITLE
No intellisense when using Clang-CL 15 with /WX

### ISSUE THREAD

**Issue by dgehriger** [microsoft/vscode-cpptools#10221]
### Environment

- OS and Version: Windows 10
- VS Code Version: 1.73.1
- C/C++ Extension Version: v1.13.5

Unlike previous versions, when running clang-cl.exe version 15.x as follows from a cmd.exe prompt:

```
clang-cl /WX /clang:-xc /clang:-E /clang:-dM nul
```

It outputs the following warning:

```
clang-cl: error: '-x c' after last input file has no effect [-Werror,-Wunused-command-line-argument]
```

Expected outcome is no error, and dumping of preprocessor variables.

Among other things, this prevents clang-cl 15.x from being used by Visual Studio Code's C++ extension for Intellisense.

### Bug Summary and Steps to Reproduce

Use clang-cl version 15 and observe missing Intellisense

### Expected behavior

Intellisense working

### Code sample and Logs

```shell
N/A
```

**Comment by sean-mcmanus** [comment]
Thanks for reporting this.

**Comment by dgehriger** [comment]
See also https://github.com/llvm/llvm-project/issues/59307

**Comment by sean-mcmanus** [comment]
Can you remove the "/WX" argument? That is unneeded when configuring IntelliSense.

**Comment by sean-mcmanus** [comment]
We have a potential fix (just skipping the /WX which we already do for the clang equivalent -Werror).

**Comment by sean-mcmanus** [comment]
The fix is available with 1.14.0 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.14.0
